File: python/with_java/programmers/level 1/추억 점수.py
def solution(name, yearning, photo):
    score = {}
    answer = []
    for i in range(len(name)):
        score[name[i]] = yearning[i]
    
    for names in photo:
        result = 0
        for n in names:
            result += score.get(n, 0)
        answer.append(result)
    return answer

File: java/src/main/java/ps_traning/programmers/level_1/추억_점수.java
package ps_traning.programmers.level_1;

import java.util.HashMap;
import java.util.Map;

public class 추억_점수 {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        Map<String, Integer> score = new HashMap<>();
        for (int i = 0; i < name.length; i++) {
            score.put(name[i], yearning[i]);
        }
        for (int i = 0; i < photo.length; i++) {
            String[] names = photo[i];
            int result = 0;
            for (String n : names) {
                result += score.getOrDefault(n, 0);
            }
            answer[i] = result;
        }
        return answer;
    }
}
